cmake/EnableWarnings.cmake

```diff
@@ -39,11 +41,18 @@ if(${ENABLE_WARNINGS})
 -Wstack-protector;\
 -Wswitch-default;\
 -Wunreachable-code;\
--Wwrite-strings")
+-Wwrite-strings" SpectreWarnings0)
```

Comment:
what is SpectreWarnings0?

Reply:
I needed a name, this seemed not too insane so we gather the warnings in `SpectreWarnings`. Because of the particular way this file is structured `SpectreWarnings` would also work if you prefer

Reply:
well my first thought on reading it was that you had a typo...  if it is the same as `SpectreWarnings` and you can reuse the name, then that is fine; otherwise prepend with `Gathered` if that is appropriate...